src/ui/public/directives/rows.js

```diff
@@ -40,10 +40,10 @@ module.directive('kbnRows', function ($compile, $rootScope, getAppState, Private
         let $cellContent;
 
         if (contents instanceof AggConfigResult) {
+          const field = contents.aggConfig.getField();
           const isCellContentFilterable =
-            contents.type === 'bucket'
-            && contents.aggConfig.getField()
-            && contents.aggConfig.getField().filterable;
+            _.get(contents, 'aggConfig.type.createFilter')
+            && (!field || field.filterable);
 
```

Comment:
I'd consider moving this check to the `AggConfig` type instead. Then you could do something like `if (aggConfig.isFilterable())`, and we don't need to ducktype this here.